### PR TITLE
chore: update java version

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: <link>

### Overview

- Update Java version and Jenkins version to LTS

### Rationale

- Java EOL support for Jenkins
- Jenkins new LTS release

### Juju Events Changes

None

### Module Changes

None

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

Some tests are broken due to: https://github.com/canonical/microk8s/issues/4765

### CODE REVIEW

**Comment by yanksyoon** [comment]
Currently the auth_proxy integration tests are broken due to: https://github.com/canonical/microk8s/issues/4765

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -794,66 +789,129 @@ async def traefik_application_fixture(model: Model):
...
     )
-
-    # See https://github.com/canonical/kratos-operator/issues/182
-    await application.model.wait_for_idle(
-        status="active",
-        apps=[application.name, oathkeeper.name] + IDENTITY_PLATFORM_APPS,
-        raise_on_error=False,
-        timeout=30 * 60,
-        idle_period=5,
+    hydra_app = await model.deploy("hydra", channel="edge", series="jammy", trust=True)
```
Comment (yanksyoon) [inline tests/integration/conftest.py:815]: Re-written due to race condition in IDP charms.
https://github.com/canonical/kratos-external-idp-integrator/issues/149

File: tests/integration/conftest.py
Comment (yanksyoon) [inline tests/integration/conftest.py:1]: Minor stylistic changes and refactoring.

Comment (yanksyoon) [reply tests/integration/conftest.py:1]: `application.model.relate` -> `model.integrate`

File: tests/integration/helpers.py [hunk 2]
```diff
@@ -51,14 +51,13 @@ async def install_plugins(
     assert res.status_code == 200, "Failed to request plugins install"
 
     # block until the UI does not have "Pending" in download progress column.
-    await unit.model.block_until(
+    await wait_for(
```
Comment (yanksyoon) [inline tests/integration/helpers.py:54]: `model.block_until` causes a lot of issues with websocket.disconnect errors. This juju interaction is extremely flaky and the custom `wait_for` should be used whenever applicable.

File: tests/integration/test_auth_proxy.py [hunk 3]
```diff
@@ -150,11 +176,15 @@ async def test_auth_proxy_integration_authorized(
...
     assert: the browser is redirected to the Jenkins URL with response code 200
     """
-    status = await application.model.get_status()
-    address = status["applications"]["traefik-public"]["public-address"]
+    unit_status = await get_application_unit_status(
+        model=application.model, application="traefik-public"
+    )
+    workload_message = str(cast(DetailedStatus, unit_status.workload_status).info)
+    # The message is: Serving at <external loadbalancer IP>
+    address = workload_message.removeprefix("Serving at ")
```
Comment (yanksyoon) [inline tests/integration/test_auth_proxy.py:187]: A proper parsing of the loadbalancer IP should be from Kubernetes API.

File: tests/integration/test_plugins_part_1.py [hunk 4]
```diff
@@ -404,243 +399,3 @@ async def test_bzr_plugin(unit_web_client: UnitWebClient):
 
     config_page = str(res.content, "utf-8")
     assert "Bazaar" in config_page, f"Bzr configuration option not found. {config_page}"
-
```
Comment (yanksyoon) [inline tests/integration/test_plugins_part_1.py:407]: Tests have been split due to resource consumption.

File: tests/integration/test_plugins_part_2.py
Comment (yanksyoon) [inline tests/integration/test_plugins_part_2.py:1]: The split test has been moved here.

File: tox.ini [hunk 5]
```diff
@@ -127,10 +130,10 @@ commands =
...
     pytest --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs}
     # uncomment the following lines to use with output of tox -e build.
-    # --jenkins-image=localhost:32000/jenkins:test \
-    # --charm-file=./jenkins-k8s_ubuntu-22.04-amd64.charm \
-    # --kube-config=~/.kube/config \
-    # --keep-models
+    ; --jenkins-image=localhost:32000/jenkins:test \
+    ; --charm-file=./jenkins-k8s_ubuntu-22.04-amd64.charm \
+    ; --kube-config=~/.kube/config \
+    ; --keep-models
```
Comment (yanksyoon) [inline tox.ini:136]: Prefer `;` commenting over `#` for ini configurations.

**Review by yanksyoon — COMMENTED** [review]
Comments to facilitate reviews

**Review by arturo-seijas — APPROVED** [review]

**Review by weiiwang01 — APPROVED** [review]

**Pull request merged**